app/models/lesson.rb

```diff
@@ -30,6 +38,16 @@ def import_content_from_github
     failed_to_import_message
   end
 
+  def has_submission?
+    is_project? &&
+    accepts_submission? &&
+    is_not_a_ruby_project? # should be removed after revamping ruby lessons
+  end
+
+  def has_live_preview?
+    has_submission? && is_not_a_ruby_project?
```

Comment:
It might look superfluous to call `is_not_a_ruby_project?` in `has_live_preview?` method as the `has_submission?` method already calls it.

We'll be removing `is_not_a_ruby_project?` method from `has_submission?` once the following multi-project lessons are split:

- [Building blocks](http://www.theodinproject.com/courses/ruby-programming/lessons/building-blocks)
- [Advanced Building blocks](http://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks)
- [OOP](http://www.theodinproject.com/courses/ruby-programming/lessons/oop)
- [File I/O and Serialization](http://www.theodinproject.com/courses/ruby-programming/lessons/file-i-o-and-serialization)
- [Recursion](http://www.theodinproject.com/courses/ruby-programming/lessons/recursion) - _Not sure if this needs to be split_
- [Data Structures And Algorithms](http://www.theodinproject.com/courses/ruby-programming/lessons/data-structures-and-algorithms)
- [Testing Ruby](http://www.theodinproject.com/courses/ruby-programming/lessons/testing-ruby)

/cc @KevinMulhern